File: OOP based implmentation/OOP based implmentation/Project/objs/Company.cpp
#include <string>
#include "Company.h"

using namespace std;

Company::Company(const string &username, const string &password, const string &name, const string &location, const string &type, int minCtc, int maxCtc)
    : CompanySignup(username, password), name(name), location(location), type(type), minCtc(minCtc), maxCtc(maxCtc)
{
}

string Company::getName() const
{
    return name;
}

string Company::getLocation() const
{
    return location;
}

string Company::getType() const
{
    return type;
}

int Company::getMinCtc() const
{
    return minCtc;
}

int Company::getMaxCtc() const
{
    return maxCtc;
}

File: OOP based implmentation/OOP based implmentation/Project/objs/Company.java
package objs;

import login.CompanySignup;

public class Company extends CompanySignup {
    private String name;
    private String location;
    private String type;
    private int minCtc;
    private int maxCtc;

    public Company(String username, String password, String name, String location, String type, int minCtc, int maxCtc) {
        super(username, password);
        this.name = name;
        this.location = location;
        this.type = type;
        this.minCtc = minCtc;
        this.maxCtc = maxCtc;
    }
    public String getName() {
        return name;
    }

    public String getLocation() {
        return location;
    }

    public String getType() {
        return type;
    }

    public int getMinCtc() {
        return minCtc;
    }

    public int getMaxCtc() {
        return maxCtc;
    }

}
